dashboard/src/components/DeploymentForm/DeploymentForm.tsx

```diff
@@ -191,7 +217,37 @@ export default function DeploymentForm() {
                     value={releaseName}
                     required={true}
                   />
+                  <CdsControlMessage error="valueMissing">
+                    A descriptive name for this application
+                  </CdsControlMessage>
                 </CdsInput>
+                {
+                  // TODO(agamez): let plugins define their own components instead of hardcoding the logic here
+                  getPluginsRequiringSA().includes(pluginObj.name) ? (
+                    <>
+                      <CdsSelect layout="horizontal" id="serviceaccount-selector">
+                        <label>Service Account</label>
+                        <select
+                          value={reconciliationOptions.serviceAccountName}
+                          onChange={onChangeSA}
+                          required={true}
+                        >
+                          <option key=""></option>
+                          {serviceAccountList?.map(o => (
+                            <option key={o} value={o}>
+                              {o}
+                            </option>
+                          ))}
+                        </select>
+                        <CdsControlMessage error="valueMissing">
+                          The Service Account name this application will be installed with.
+                        </CdsControlMessage>
```

Comment:
Using CDS components brings some form validation capabilities out-of-the-box (idem with inputs).